crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -852,6 +852,89 @@ impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::PaymentsPostSessionTo
     }
 }
 
+#[cfg(feature = "v1")]
+#[async_trait]
+impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::PaymentsUpdateMetadataData>
+    for PaymentResponse
+{
+    async fn update_tracker<'b>(
+        &'b self,
+        db: &'b SessionState,
+        mut payment_data: PaymentData<F>,
+        router_data: types::RouterData<
+            F,
+            types::PaymentsUpdateMetadataData,
+            types::PaymentsResponseData,
+        >,
+        key_store: &domain::MerchantKeyStore,
+        storage_scheme: enums::MerchantStorageScheme,
+        _locale: &Option<String>,
+        #[cfg(all(feature = "v1", feature = "dynamic_routing"))] _routable_connector: Vec<
+            RoutableConnectorChoice,
+        >,
+        #[cfg(all(feature = "v1", feature = "dynamic_routing"))]
+        _business_profile: &domain::Profile,
+    ) -> RouterResult<PaymentData<F>>
+    where
+        F: 'b + Send,
+    {
+        let connector = payment_data
+            .payment_attempt
+            .connector
+            .clone()
+            .ok_or(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("connector not found")?;
+
+        match router_data.response.clone() {
+            Ok(types::PaymentsResponseData::SessionUpdateResponse { status, .. }) => {
+                if status == SessionUpdateStatus::Success {
+                    let m_db = db.clone().store;
+                    let payment_intent = payment_data.payment_intent.clone();
+                    let key_manager_state: KeyManagerState = db.into();
+
+                    let payment_intent_update =
+                            hyperswitch_domain_models::payments::payment_intent::PaymentIntentUpdate::MetadataUpdate {
+                                metadata: payment_data.payment_intent.metadata.clone()
+                                          .ok_or(errors::ApiErrorResponse::InternalServerError)
+                                          .attach_printable("payment_intent.metadata not found")?,
+                                updated_by:payment_data.payment_intent.updated_by.clone(), };
```

Comment:
Nit (formatting): Just copy-pasted this snippet in the Rust playground to format it.

```suggestion
                    let payment_intent_update =
                        hyperswitch_domain_models::payments::payment_intent::PaymentIntentUpdate::MetadataUpdate {
                            metadata: payment_data
                                .payment_intent
                                .metadata
                                .clone()
                                .ok_or(errors::ApiErrorResponse::InternalServerError)
                                .attach_printable("payment_intent.metadata not found")?,
                            updated_by: payment_data.payment_intent.updated_by.clone(),
                        };
```